File: src/DynamicLinkLibraries/Vector3D/EulerAngles.cs

namespace Vector3D
{
    /// <summary>
    /// Euler angles
    /// </summary>
    public class EulerAngles
    {
        /// <summary>
        /// Default constructor
        /// </summary>
        public EulerAngles()
        {
            roll = 0;
            pitch = 0;
            yaw = 0;
        }

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="roll">Roll</param>
        /// <param name="pitch">Pitch</param>
        /// <param name="yaw">Yaw</param>
        public EulerAngles(double  roll, double pitch, double yaw)
        {
            this.roll = roll;
            this.pitch = pitch;
            this.yaw = yaw;
        }

        /// <summary>
        /// Roll 
        /// </summary>
        public double roll { get; set; }

        /// <summary>
        /// Pitch
        /// </summary>
        public double pitch { get; set; }

        /// <summary>
        /// Yaw
        /// </summary>
        public double yaw { get; set; }

 
    }
}

File: src/TypeScript/TypeScriptLibrary/src/Library/Vector3D/EulerAngles.ts
export class EulerAngles {

    roll: number = 0;
    pitch: number = 0;
    yaw: number = 0;

    constructor(roll: number, pitch: number, yaw: number) {
        this.roll = roll;
        this.pitch = pitch
        this.yaw = yaw
    }

    getRoll(): number {
        return this.roll
    }

    getPitch(): number {
        return this.pitch
    }

    getYaw(): number {
        return this.yaw
    }

    setRoll(roll: number): void {
        this.roll = roll
    }

    setPitch(pitch: number): void {
        this.pitch = pitch
    }

    setYaw(yaw: number): void {
        this.yaw = yaw
    }
}
